RELEASE.md

```diff
@@ -0,0 +1,9 @@
+## Release Finalization
+
+To cut a release:
+- Create a release branch in the format `release/0.99.0`
+  - New commits to this branch will trigger the `build` workflow and produce a lifecycle image: `buildpacksio/lifecycle:<commit sha>`
```

Comment:
```suggestion
1. Create a release branch in the format `release/0.99.0`. New commits to this branch will trigger the `build` workflow and produce a lifecycle image: `buildpacksio/lifecycle:<commit sha>`
```

Reply:
I found this a bit easier to read than the nested bullet points. Happy to leave it how it is if you feel strongly about it, but I thought I'd throw it out there.